src/jitter.rs

```diff
@@ -126,28 +154,34 @@ impl From<TimerError> for Error {
 #[cfg(feature="std")]
 static JITTER_ROUNDS: AtomicUsize = ATOMIC_USIZE_INIT;
 
-impl JitterRng {
+#[cfg(feature="std")]
+impl JitterRng<Timer> {
     /// Create a new `JitterRng`.
     /// Makes use of `std::time` for a timer.
     ///
     /// During initialization CPU execution timing jitter is measured a few
     /// hundred times. If this does not pass basic quality tests, an error is
     /// returned. The test result is cached to make subsequent calls faster.
-    #[cfg(feature="std")]
-    pub fn new() -> Result<JitterRng, TimerError> {
-        let mut ec = JitterRng::new_with_timer(platform::get_nstime);
-        let mut rounds = JITTER_ROUNDS.load(Ordering::Relaxed) as u32;
+    pub fn new() -> Result<JitterRng<Timer>, TimerError> {
+        let mut state = JitterRng::new_with_timer(Timer::default());
+
+        let mut rounds = JITTER_ROUNDS.load(Ordering::Relaxed) as u8;
         if rounds == 0 {
             // No result yet: run test.
             // This allows the timer test to run multiple times; we don't care.
-            rounds = ec.test_timer()?;
+            rounds = state.test_timer()?;
             JITTER_ROUNDS.store(rounds as usize, Ordering::Relaxed);
             info!("JitterRng: using {} rounds per u64 output", rounds);
         }
-        ec.set_rounds(rounds);
-        Ok(ec)
+        state.set_rounds(rounds);
+
+        // Fill `data` with a non-zero value.
+        state.gen_entropy();
```

Comment:
Why bother? Doesn't look like it gets used.

Reply:
It gets used, see https://github.com/rust-lang-nursery/rand/pull/251#discussion_r165816872